v3/src/components/graph/models/graph-controller.ts

```diff
@@ -113,32 +113,35 @@ export class GraphController {
     })
     graphModel.setPlotType(plotChoices[attrTypes.x][attrTypes.y]);
     ['x', 'y'].forEach((attrPlace: GraphAttrRole) => {
-      const axisPlace = attrPlaceToAxisPlace[attrPlace] as AxisPlace,
+      const axisPlace = attrRoleToAxisPlace[attrPlace],
         attrType = attrTypes[attrPlace]
-      let axisModel
-      switch (attrType) {
-        case 'numeric':
-          axisModel = NumericAxisModel.create({place: axisPlace, min: 0, max: 1})
-          graphModel.setAxis(axisPlace, axisModel)
-          layout.setAxisScale(axisPlace, scaleLinear())
-          setNiceDomain(dataConfig.numericValuesForPlace(attrPlace), axisModel)
-          break
-        case 'categorical':
-          axisModel = CategoricalAxisModel.create({place: axisPlace})
-          graphModel.setAxis(axisPlace, axisModel)
-          layout.setAxisScale(axisPlace,
-            scaleBand().domain(dataConfig.categorySetForPlace(attrPlace)))
-          break
-        default:
-          axisModel = EmptyAxisModel.create({place: axisPlace})
-          graphModel.setAxis(axisPlace, axisModel)
-          layout.setAxisScale(axisPlace, scaleOrdinal())
+      if(axisPlace) {
+        let axisModel
+        switch (attrType) {
+          case 'numeric':
+            axisModel = NumericAxisModel.create({place: axisPlace, min: 0, max: 1})
+            graphModel.setAxis(axisPlace, axisModel)
+            layout.setAxisScale(axisPlace, scaleLinear())
+            setNiceDomain(dataConfig.numericValuesForPlace(attrPlace), axisModel)
+            break
+          case 'categorical':
+            axisModel = CategoricalAxisModel.create({place: axisPlace})
+            graphModel.setAxis(axisPlace, axisModel)
+            layout.setAxisScale(axisPlace,
+              scaleBand().domain(dataConfig.categorySetForPlace(attrPlace)))
+            break
+          default:
+            axisModel = EmptyAxisModel.create({place: axisPlace})
+            graphModel.setAxis(axisPlace, axisModel)
+            layout.setAxisScale(axisPlace, scaleOrdinal())
+        }
       }
     })
   }
 
   handleAttributeAssignment(axisPlace: AxisPlace, attrID: string) {
```

Comment:
Should this take a GraphPlace instead of an AxisPlace?

Reply:
No because we're only dealing with ['x', 'y'] for now.